packages/istanbul-lib-instrument/src/instrumenter.js

```diff
@@ -80,7 +80,13 @@ class Instrumenter {
         const opts = this.opts;
         const ast = babylon.parse(code, {
             allowReturnOutsideFunction: opts.autoWrap,
-            sourceType: opts.esModules ? "module" : "script"
+            sourceType: opts.esModules ? "module" : "script",
+            plugins: [
+              'asyncGenerators',
+              'dynamicImport',
+              'flow',
+              'jsx'
+            ]
```

Comment:
@Axbon @jonathanargentiero if you don't beat me too it, I'm hoping to do a bit of coding this afternoon and will see if I can't get a test around this -- something testing `jsx` seems like it would be a good exa,mple in the codebase.